openshift/openshift-ansible-contrib
# Add Extra Docker Registry URLs (custom post-provision action)

#### What does this PR do?
This is another example playbook added to the custom-actions directory and documented in the provisioning READMe.

#### How should this be manually tested?
1. Run the provisioning.
2. Connect to an arbitrary host in your cluster and display the content of the `/etc/containers/registries.conf` file. Optionally, add a line with `registries: something` or a list of random items.
3. Run the add-docker-registry.yml playbook like so:
```
ansible-playbook -i <inventory> openshift-ansible-contrib/playbooks/provisioning/openstack/custom-actions/add-docker-registry.yml  --extra-vars '{"registries": "reg1", "insecure_registries": ["ins_reg1","ins_reg2"]}'
```
4. Check that the playbook finished successfully.
5. Connect to an arbitrary host in your cluster and display the content of the `/etc/containers/registries.conf` file again. Check that it includes the newly added registries as well as the old ones (if there were any).
6. Run `docker info` as root and check that these registries were added (under `Registries` and `Insecure Registries`).

#### Is there a relevant Issue open for this?
Provide a link to any open issues that describe the problem you are solving.

#### Who would you like to review this?
cc: @tomassedovic @bogdando @tzumainn PTAL
